app/src/main/java/eu/kanade/domain/chapter/interactor/SyncChaptersWithSource.kt

```diff
@@ -206,8 +215,16 @@ class SyncChaptersWithSource(
 
         val excludedScanlators = getExcludedScanlators.await(manga.id).toHashSet()
 
-        return updatedToAdd.filterNot {
+        val nonExcludedChapters = updatedToAdd.filterNot {
             it.url in reAddedUrls || it.scanlator in excludedScanlators
         }
+
+        val chaptersToDownload = if (downloadPreferences.downloadUnreadChaptersOnly().get()) {
+            nonExcludedChapters.filter { getReadChapterCount.await(manga.id, it.chapterNumber) == 0L }
```

Comment:
You already have access to db chapters here why are you making more db requests?